Content.Shared/Clothing/EntitySystems/MaskSystem.cs

```diff
@@ -64,10 +67,18 @@ private void OnGotUnequipped(EntityUid uid, MaskComponent mask, GotUnequippedEve
 
     private void ToggleMaskComponents(EntityUid uid, MaskComponent mask, EntityUid wearer, bool isEquip = false)
     {
-        var maskEv = new ItemMaskToggledEvent(wearer, mask.IsToggled, isEquip);
+        var maskEv = new ItemMaskToggledEvent(wearer, equippedPrefix: "toggled", mask.IsToggled, isEquip);
         RaiseLocalEvent(uid, ref maskEv);
 
         var wearerEv = new WearerMaskToggledEvent(mask.IsToggled);
         RaiseLocalEvent(wearer, ref wearerEv);
     }
+
+    private void OnGotFolded(Entity<MaskComponent> ent, ref GotFoldedEvent args)
+    {
+        ent.Comp.IsToggled = args.IsFolded;
+
+        var wearerEv = new ItemMaskToggledEvent(ent.Owner, equippedPrefix: null, ent.Comp.IsToggled, args.IsFolded);
+        RaiseLocalEvent(ent.Owner, ref wearerEv);
+    }
 }
```

Comment:
Split it out to its own method rather than copy-pasting from one method to another.

Reply:
I modified ToggleMaskComponents slightly so that I can just call it from OnFolded. I'm passing the bandana's ID as the 'wearer', but that doesn't really matter is this can't be folded while worn anyways. (I don't love doing that, but the event needs to be called to reset some of the other components.)